File: src/main/java/org/shark/example/service/kafka/PushKafkaJsonQueueService.java
package org.shark.example.service.kafka;

import lombok.RequiredArgsConstructor;
import org.shark.example.service.base.pojo.ResponseDto;
import org.shark.example.service.rabbitmq.pojo.MessageQueueDto;
import org.shark.example.service.rabbitmq.pojo.MessageQueueInputDto;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@RequiredArgsConstructor
@Service
public class PushKafkaJsonQueueService {

    private final KafkaTemplate<String, MessageQueueDto> kafkaTemplate;

    public ResponseDto<Void> start(MessageQueueInputDto messageQueueInputDto) {
        MessageQueueDto messageQueueDto = new MessageQueueDto();
        messageQueueDto.setData(messageQueueInputDto.getData());
        messageQueueDto.setUuid(UUID.randomUUID().toString());
        kafkaTemplate.send("json_topic", messageQueueDto);
        return ResponseDto.<Void>builder().status(true).build();
    }
}

File: src/main/kotlin/org/shark/example/service/kafka/PushKafkaJsonQueueService.kt
package org.shark.example.service.kafka

import org.shark.example.service.base.pojo.ResponseDto
import org.shark.example.service.rabbitmq.pojo.MessageQueueDto
import org.shark.example.service.rabbitmq.pojo.MessageQueueInputDto
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service
import java.util.*

@Service
class PushKafkaJsonQueueService(
    private val kafkaTemplate: KafkaTemplate<String, MessageQueueDto>
) {
    fun start(messageQueueInputDto: MessageQueueInputDto): ResponseDto<Void> {
        val messageQueueDto = MessageQueueDto(
            data = messageQueueInputDto.data,
            uuid = UUID.randomUUID().toString()
        )
        kafkaTemplate.send("json_topic", messageQueueDto)
        return ResponseDto(status = true)
    }
}
